downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/ProtegeExporter.java

```diff
@@ -0,0 +1,404 @@
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ForkJoinPool;
+import java.util.stream.Collectors;
+
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+
+/**
+ * Exports top-level pathways in Protege format, using the Perl code in GKB::WebUtils, and then
+ * packages the resulting files into a single tar file.
+ * Takes ~1.5 hours on my (Solomon) workstation, with 7 threads running 8 cores. Of course, it ran the system
+ * at a very high load. 5 threads (concurrent protege exporters) finished in almost the same amount of time
+ * and the system was not too overloaded. Less than that seemed to complete slower.
+ * @author sshorser
+ *
+ */
+public class ProtegeExporter
+{
+	// The code in GKB::WebUtils always writes protege files to /tmp/ and it doesn't look like that is configurable,
+	// so we'll work in /tmp as well.
+	private static final String PROTEGE_ARCHIVE_PATH = "/tmp/protege_files.tar";
+	private static final String PROTEGE_FILES_DIR = PROTEGE_ARCHIVE_PATH.replace(".tar", "/");
+	private static final Logger logger = LogManager.getLogger();
+	private String releaseDirectory;
+	private int parallelism = ForkJoinPool.getCommonPoolParallelism();
+	private List<String> extraIncludes = new ArrayList<>();
+	private String pathToWrapperScript = "";
+	private String downloadDirectory;
+	private Set<Long> pathwayIdsToProcess = new HashSet<>();
+	private Set<String> speciesToProcess = new HashSet<>();
+	
+	public ProtegeExporter()
+	{
+		// Default no-arg consturctor.
+	}
+	
+	public ProtegeExporter(Properties props, String releaseDir, String downloadDir)
+	{
+		this.setReleaseDirectory(releaseDir);
+		String propsPrefix = "protegeexporter";
+		int degP = Integer.parseInt(props.getProperty(propsPrefix + ".parallelism"));
+		this.setParallelism(degP);
+		String extraIncludeStr = props.getProperty(propsPrefix+".extraIncludes");
+		if (extraIncludeStr != null && !extraIncludeStr.trim().isEmpty())
+		{
+			List<String> extraIncs = Arrays.asList(extraIncludeStr.split(","));
+			this.setExtraIncludes(extraIncs);
+		}
+		String pathToWrapper = props.getProperty(propsPrefix+".pathToWrapperScript");
+		this.setPathToWrapperScript(pathToWrapper);
+		this.setDownloadDirectory(downloadDir);
+		
+		String filterIds = props.getProperty(propsPrefix+".filterIds");
+		if (filterIds != null && !filterIds.trim().isEmpty())
+		{
+			List<String> ids = Arrays.asList(filterIds.split(","));
+			this.setPathwayIdsToProcess( ids.stream().map(Long::valueOf).collect(Collectors.toSet()) );
+		}
+		
+		String filterSpecies = props.getProperty(propsPrefix+".filterSpecies");
+		if (filterSpecies != null && !filterSpecies.trim().isEmpty())
```

Comment:
Could this be a helper function since it is used three times?  Something like:
```
private boolean hasNonWhiteSpace(String str) {
    return str != null && !str.trim().isEmpty();
}
```
There's probably a better method name :)

Reply:
It _could_ be but I don't think it's worth it in this case.

Reply:
Okay, just a thought.